src/runtime/wasm.ts

```diff
@@ -199,26 +251,29 @@ export class WasmParticle extends Particle {
   async initialize(buffer: ArrayBuffer) {
     assert(this.spec.name.length > 0);
 
-    // TODO: detect errors when this memory size doesn't match up with the wasm's declared values?
-    this.memory = new WebAssembly.Memory({initial: 256, maximum: 256});
-    this.heap = new Uint8Array(this.memory.buffer);
+    // TODO: vet the imports/exports on 'module'
+    const module = await WebAssembly.compile(buffer);
+    const emc = readEmscriptenMetadata(module);
+
+    this.memory = new WebAssembly.Memory({initial: emc.memSize, maximum: emc.memSize});
```

Comment:
The Kotlin bootstrap just uses the default memory (instance.exports.memory.buffer) and doesn't try to create memory itself for import. I'm hoping this just works and Kotlin doesn't care.

Reply:
There's no such thing as "default memory" in the C++/emscripten world, and since you're referencing one from instance.__exports__ it looks like Kotlin's build is both setting up the memory buffer and exporting it for the driver code to access. That's the opposite of emscripten's approach.